in_toto/runlib.py

```diff
@@ -249,76 +231,11 @@ def record_artifacts_as_dict(artifacts, exclude_patterns=None,
-    if os.path.isfile(artifact):
-      # FIXME: this is necessary to provide consisency between windows
-      # filepaths and *nix filepaths. A better solution may be in order
-      # though...
-      artifact = artifact.replace('\\', '/')
-      key = _apply_left_strip(artifact, artifacts_dict, lstrip_paths)
-      artifacts_dict[key] = _hash_artifact(artifact,
-          normalize_line_endings=normalize_line_endings)
-
-    elif os.path.isdir(artifact):
-      for root, dirs, files in os.walk(artifact,
-          followlinks=follow_symlink_dirs):
-        # Create a list of normalized dirpaths
-        dirpaths = []
-        for dirname in dirs:
-          norm_dirpath = os.path.normpath(os.path.join(root, dirname))
-          dirpaths.append(norm_dirpath)
-
-        # Applying exclude patterns on the directory paths returned by walk
-        # allows to exclude a subdirectory 'sub' with a pattern 'sub'.
-        # If we only applied the patterns below on the subdirectory's
-        # containing file paths, we'd have to use a wildcard, e.g.: 'sub*'
-        if exclude_patterns:
-          dirpaths = _apply_exclude_patterns(dirpaths, exclude_filter)
-
-        # Reset and refill dirs with remaining names after exclusion
-        # Modify (not reassign) dirnames to only recurse into remaining dirs
-        dirs[:] = []
-        for dirpath in dirpaths:
-          # Dirs only contain the basename and not the full path
-          name = os.path.basename(dirpath)
-          dirs.append(name)
-
-        # Create a list of normalized filepaths
-        filepaths = []
-        for filename in files:
-          norm_filepath = os.path.normpath(os.path.join(root, filename))
-
-          # `os.walk` could also list dead symlinks, which would
-          # result in an error later when trying to read the file
-          if os.path.isfile(norm_filepath):
-            filepaths.append(norm_filepath)
-
-          else:
-            LOG.info("File '{}' appears to be a broken symlink. Skipping..."
-                .format(norm_filepath))
-
-        # Apply exlcude patterns on the normalized file paths returned by walk
-        if exclude_patterns:
-          filepaths = _apply_exclude_patterns(filepaths, exclude_filter)
-
-        for filepath in filepaths:
-          # FIXME: this is necessary to provide consisency between windows
-          # filepaths and *nix filepaths. A better solution may be in order
-          # though...
-          normalized_filepath = filepath.replace("\\", "/")
-          key = _apply_left_strip(
-              normalized_filepath, artifacts_dict, lstrip_paths)
-          artifacts_dict[key] = _hash_artifact(filepath,
-              normalize_line_endings=normalize_line_endings)
-
-    # Path is no file and no directory
-    else:
-      LOG.info("path: {} does not exist, skipping..".format(artifact))
-
+  for artifact in resolved_artifacts:
+    key = _apply_left_strip(artifact, artifacts_dict, lstrip_paths)
```

Comment:
This should be updated to left strip while ignoring URI scheme if available.

Reply:
I think that makes sense. On the other hand, do we support lstripping paths based on its scheme? i.e., lstrip for file paths but not for scheme X which might also match the pattern. :thinking:

Reply:
For now, let's set lstrip to apply to the parts after the scheme and we can revisit it later I think.

Reply:
Let's keep URI-specific lstrips as a future issue.

Reply:
I have lstrip working on only the left part of the scheme. What I meant by URI-specific is the lstripping of paths based on scheme.

Reply:
Could you give me an example?

Reply:
If you look at the tests `TestApplyLeftStrip` it does it.

If you are asking for examples to URI-specific lstripping, I can foresee scenarios where an artifact's URI has the same/very-similar URI to a file path in a user's `lstrip_paths` but they do not want it to apply to the aforementioned artifact.

Reply:
Left strip will be handled by the resolver.